### PR TITLE
Some cgroups cleanup

### PR DESCRIPTION
**Pull request by hqhq**
- Refactor find mount point functions
- Rename FindCgroupMountpointAndSource to FindCgroupMountpointAndRoot

### CODE REVIEW

File: libcontainer/cgroups/utils.go [hunk 1]
```diff
@@ -21,49 +21,31 @@ const cgroupNamePrefix = "name="
 
 // https://www.kernel.org/doc/Documentation/cgroups/cgroups.txt
 func FindCgroupMountpoint(subsystem string) (string, error) {
-	f, err := os.Open("/proc/self/mountinfo")
+	mounts, err := mount.GetMounts()
```
Comment (mrunalp) [inline libcontainer/cgroups/utils.go]: I think the reason @LK4D4 was parsing /proc/self/mountinfo directly as it was more efficient than this call. 
ping @LK4D4

Comment (LK4D4) [inline libcontainer/cgroups/utils.go]: Yeah, it's like super-inefficient, 60% of container start was spent in that func.

Comment (hqhq) [inline libcontainer/cgroups/utils.go]: All right, then I think we should keep it efficient, but it breaks freebsd, maybe freebsd should offer an get cgroup mount point function, we can deal with that later.
I'll update this, thanks.

**Comment by hqhq** [comment]
@mrunalp @LK4D4 Updated.

**Comment by LK4D4** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**